### PR TITLE
Migrate Type Hints for PEP 585

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Update type hints following [PEP 585](https://peps.python.org/pep-0585/).

### CODE REVIEW

**Review by b-chu — APPROVED** [review]
Thanks!

File: composer/loggers/in_memory_logger.py [hunk 1]
```diff
@@ -49,32 +49,32 @@ class InMemoryLogger(LoggerDestination):
...
             a (:class:`~.time.Timestamp`, logged value) tuple.
             This dictionary contains all logged data.
-        most_recent_values (Dict[str, Any]): Mapping of a key to the most recent value for that key.
-        most_recent_timestamps (Dict[str, Timestamp]): Mapping of a key to the
+        most_recent_values (dict[str, Any]): Mapping of a key to the most recent value for that key.
+        most_recent_timestamps (dict[str, Timestamp]): Mapping of a key to the
             :class:`~.time.Timestamp` of the last logging call for that key.
-        hyperparameters (Dict[str, Any]): Dictionary of all hyperparameters.
-        tables (Dict[str, str]): Dictionary of table name to json table.
+        hyperparameters (dict[str, Any]): dictionary of all hyperparameters.
```
Comment (snarayan21) [inline composer/loggers/in_memory_logger.py:58]: These should still be capitalized no?

File: composer/utils/batch_helpers.py [hunk 2]
```diff
@@ -174,8 +174,8 @@ def _batch_set_multiple(batch: Any, key: Any, value: Any) -> Any:
 
 
 def _batch_set_tuple(batch: Any, key: Union[int, str], value: Any) -> Any:
-    """Sets key value pairs in tuples and NamedTuples."""
-    if hasattr(batch, '_fields'):  # NamedTuple
+    """Sets key value pairs in tuples and Namedtuples."""
```
Comment (snarayan21) [inline composer/utils/batch_helpers.py:177]: should still be NamedTuples here and below

File: tests/loggers/test_cometml_logger.py [hunk 3]
```diff
@@ -211,7 +211,7 @@ def test_comet_ml_logging_train_loop(monkeypatch, tmp_path):
     monkeypatch.setattr(comet_ml, 'Experiment', comet_ml.OfflineExperiment)
     from composer.loggers import CometMLLogger
 
-    # Set offline directory.
+    # set offline directory.
```
Comment (snarayan21) [inline tests/loggers/test_cometml_logger.py:214]: should still be capitalized

**Review by snarayan21 — COMMENTED** [review]
some capitalization stuff, other than that lgtm

get those lines up fr

**Review by karan6181 — APPROVED** [review]
LGTM. Thank You!

**Pull request merged**